app/models/occupation_standard.rb

```diff
@@ -377,6 +377,20 @@ def state_id
     state&.id
   end
 
+  def clean_onet_code
+    # correct format 12-3456.07
+    base_onet = onet_code
+
+    if onet_code.present?
+      return nil if onet_code == "onet"
+
+      base_onet = base_onet.strip.gsub!(/[^0-9A-Za-z]/, "")
```

Comment:
```suggestion
      base_onet = base_onet.strip.gsub(/[^0-9A-Za-z]/, "")
```
The bang replaces in place, but since you are assigning back to `base_onet`, you don't want the bang.